compiler/src/iree/compiler/Dialect/Stream/Transforms/test/refine_usage.mlir

```diff
@@ -85,11 +85,11 @@ util.func private @propagateBlocks(%cond: i1, %size: index) -> (!stream.resource
   // CHECK-SAME:               ^bb2
   cf.cond_br %cond, ^bb1(%fill0, %bb1_1_new : !stream.resource<*>, !stream.resource<*>),
                  ^bb2(%fill0, %bb1_1_new : !stream.resource<*>, !stream.resource<*>)
-// CHECK: ^bb2(%[[BB2_ARG_FILL0:.+]]: !stream.resource<{{.*}}>, %[[BB2_ARG_SELECT:.+]]: !stream.resource<{{.*}}>)
+// CHECK: ^bb2
 ^bb2(%bb2_0: !stream.resource<*>, %bb2_1: !stream.resource<*>):
   // CHECK-NOT: stream.async.transfer
   %external_transfer = stream.async.transfer %bb2_1 : !stream.resource<*>{%size} -> !stream.resource<external>{%size}
-  // CHECK: util.return %[[BB2_ARG_FILL0]], %[[BB2_ARG_SELECT]] : !stream.resource<transient>, !stream.resource<external>
+  // CHECK: util.return %[[FILL0]], %[[SELECT]] : !stream.resource<transient>, !stream.resource<external>
```

Comment:
I confirmed that https://github.com/llvm/llvm-project/commit/441b672bbdc68ad88036f3e258759854c8283adb breaks the test. The new output looks okay to me..

Input IR:

```mlir
util.func private @propagateBlocks(%cond: i1, %size: index) -> (!stream.resource<*>, !stream.resource<external>) {
  %c0 = arith.constant 0 : index
  %c128 = arith.constant 128 : index
  %c123_i32 = arith.constant 123 : i32
  %c456_i32 = arith.constant 456 : i32
  %splat0 = stream.async.splat %c123_i32 : i32 -> !stream.resource<*>{%size}
  %splat1 = stream.async.splat %c456_i32 : i32 -> !stream.resource<*>{%size}
  cf.br ^bb1(%splat0, %splat1 : !stream.resource<*>, !stream.resource<*>)
^bb1(%bb1_0: !stream.resource<*>, %bb1_1: !stream.resource<*>):
  %clone0 = stream.async.clone %bb1_0 : !stream.resource<*>{%size} -> !stream.resource<*>{%size}
  %fill0 = stream.async.fill %c123_i32, %clone0[%c0 to %c128 for %c128] : i32 -> !stream.resource<*>{%size}
  %clone1 = stream.async.clone %bb1_1 : !stream.resource<*>{%size} -> !stream.resource<*>{%size}
  %fill1 = stream.async.fill %c456_i32, %clone1[%c0 to %c128 for %c128] : i32 -> !stream.resource<*>{%size}
  %bb1_1_new = arith.select %cond, %splat1, %fill1 : !stream.resource<*>
  cf.cond_br %cond, ^bb1(%fill0, %bb1_1_new : !stream.resource<*>, !stream.resource<*>),
                 ^bb2(%fill0, %bb1_1_new : !stream.resource<*>, !stream.resource<*>)
^bb2(%bb2_0: !stream.resource<*>, %bb2_1: !stream.resource<*>):
  %external_transfer = stream.async.transfer %bb2_1 : !stream.resource<*>{%size} -> !stream.resource<external>{%size}
  util.return %bb2_0, %external_transfer : !stream.resource<*>, !stream.resource<external>
}
```

The original output:

```mlir
  util.func private @propagateBlocks(%arg0: i1, %arg1: index) -> (!stream.resource<transient>, !stream.resource<external>) {
    %c0 = arith.constant 0 : index
    %c128 = arith.constant 128 : index
    %c123_i32 = arith.constant 123 : i32
    %c456_i32 = arith.constant 456 : i32
    %0 = stream.async.splat %c123_i32 : i32 -> !stream.resource<transient>{%arg1}
    %1 = stream.async.splat %c456_i32 : i32 -> !stream.resource<external>{%arg1}
    cf.br ^bb1(%0, %1 : !stream.resource<transient>, !stream.resource<external>)
  ^bb1(%2: !stream.resource<transient>, %3: !stream.resource<external>):  // 2 preds: ^bb0, ^bb1
    %4 = stream.async.clone %2 : !stream.resource<transient>{%arg1} -> !stream.resource<transient>{%arg1}
    %5 = stream.async.fill %c123_i32, %4[%c0 to %c128 for %c128] : i32 -> %4 as !stream.resource<transient>{%arg1}
    %6 = stream.async.clone %3 : !stream.resource<external>{%arg1} -> !stream.resource<external>{%arg1}
    %7 = stream.async.fill %c456_i32, %6[%c0 to %c128 for %c128] : i32 -> %6 as !stream.resource<external>{%arg1}
    %8 = arith.select %arg0, %1, %7 : !stream.resource<external>
    cf.cond_br %arg0, ^bb1(%5, %8 : !stream.resource<transient>, !stream.resource<external>), ^bb2(%5, %8 : !stream.resource<transient>, !stream.resource<external>)
  ^bb2(%9: !stream.resource<transient>, %10: !stream.resource<external>):  // pred: ^bb1
    util.return %9, %10 : !stream.resource<transient>, !stream.resource<external>
  }
```

New output:

```mlir
  util.func private @propagateBlocks(%arg0: i1, %arg1: index) -> (!stream.resource<transient>, !stream.resource<external>) {
    %c0 = arith.constant 0 : index
    %c128 = arith.constant 128 : index
    %c123_i32 = arith.constant 123 : i32
    %c456_i32 = arith.constant 456 : i32
    %0 = stream.async.splat %c123_i32 : i32 -> !stream.resource<transient>{%arg1}
    %1 = stream.async.splat %c456_i32 : i32 -> !stream.resource<external>{%arg1}
    cf.br ^bb1(%0, %1 : !stream.resource<transient>, !stream.resource<external>)
  ^bb1(%2: !stream.resource<transient>, %3: !stream.resource<external>):  // 2 preds: ^bb0, ^bb1
    %4 = stream.async.clone %2 : !stream.resource<transient>{%arg1} -> !stream.resource<transient>{%arg1}
    %5 = stream.async.fill %c123_i32, %4[%c0 to %c128 for %c128] : i32 -> %4 as !stream.resource<transient>{%arg1}
    %6 = stream.async.clone %3 : !stream.resource<external>{%arg1} -> !stream.resource<external>{%arg1}
    %7 = stream.async.fill %c456_i32, %6[%c0 to %c128 for %c128] : i32 -> %6 as !stream.resource<external>{%arg1}
    %8 = arith.select %arg0, %1, %7 : !stream.resource<external>
    cf.cond_br %arg0, ^bb1(%5, %8 : !stream.resource<transient>, !stream.resource<external>), ^bb2
  ^bb2:  // pred: ^bb1
    util.return %5, %8 : !stream.resource<transient>, !stream.resource<external>
  }
```

I think we can drop the bb2 arguments in this case. @benvanik does the fix make sense?